### PR TITLE
Update JWT to support per-path validation.

### PR DESCRIPTION
**Pull request by yangminzhu**

### CODE REVIEW

File: authentication/v1alpha1/policy.proto [hunk 1]
```diff
@@ -127,10 +127,9 @@ message Jwt {
   // For example, `query=jwt_token`.
   repeated string jwt_params = 7;
 
-  // URL paths that should be excluded from the JWT validation. If the request path is matched,
-  // the JWT validation will be skipped and the request will proceed regardless.
+  // URL paths that should be excluded from the JWT validation. If the request path is
+  // matched, the JWT validation will be skipped and the request will proceed regardless.
   // This is useful to keep a couple of URLs public for external health checks.
-  // Example: "/health_check", "/status/cpu_usage".
   repeated istio.common.v1alpha1.StringMatch excluded_paths = 8;
```
Comment (diemtvu) [inline authentication/v1alpha1/policy.proto]: May be we should have both `included` and `excluded` rules, and the final trigger is ORs of `included_path` AND NOT (ORs of `excluded_path`)  with exception included_path.empty() it's simply NOT (ORs of `excluded path`)  (i.e include_path.empty() is treated as match all)

File: authentication/v1alpha1/policy.proto [hunk 2]
```diff
@@ -127,10 +127,9 @@ message Jwt {
   // For example, `query=jwt_token`.
   repeated string jwt_params = 7;
 
-  // URL paths that should be excluded from the JWT validation. If the request path is matched,
-  // the JWT validation will be skipped and the request will proceed regardless.
+  // URL paths that should be excluded from the JWT validation. If the request path is
```
Comment (diemtvu) [inline authentication/v1alpha1/policy.proto]: More accurate comment could be ` If trigger condition doesn't meet, authentication will skip the underlying JWT validation'

And in the origin block, add comment to mention JWT is used only when the trigger condition match. If none of them are trigger, request will be accepted as if there is no JWT defined.

File: authentication/v1alpha1/policy.proto (hunk 1 above)
Comment (diemtvu) [reply authentication/v1alpha1/policy.proto]: We may put them inside another message, say `TriggerCondition`

**Comment by yangminzhu** [comment]
@diemtvu Please take a look, I have updated the design based on our discussions.

**Comment by yangminzhu** [comment]
@diemtvu Updated all comments, also renamed to TriggerRule, PTAL.

**Pull request merged**